chains/manager.go

```diff
@@ -596,6 +581,22 @@ func (m *manager) buildChain(chainParams ChainParameters, sb subnets.Subnet) (*c
 		return nil, errUnknownVMType
 	}
 
+	snowmanMetrics, err := metrics.MakeAndRegister(
+		m.snowmanGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	vmMetrics, err := m.getOrMakeVMRegisterer(chainParams.VMID, primaryAlias)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
The previous code makes and registers the metrics prior to `vm.Initialize(...)`. The new code makes and registers after `vm.Initialize` has been called, but `Initialize` assumes that it has been set correctly.

The alternative would be to create a non-empty value, but only register it to the parent so that it starts ingesting metrics from it after `Initialize` has completed.